UndertaleModLib/Compiler/Parser.cs

```diff
@@ -682,7 +691,17 @@ private static Statement ParseFunction(CompileContext context)
                 if (EnsureTokenKind(TokenKind.CloseParen) == null) return null;
 
                 result.Children.Add(ParseStatement(context));
-                return result;
+                if (expression)
+                    return result;
+                else // Whatever you call non-anonymous
```

Comment:
Else is unnecessary here, due to the return from above.

Reply:
I feel that keeping both returns on the same level makes it clear at a glance that neither is especially a "default". You can't miss the if anymore and assume it always does the one thing.

Reply:
Which is fair, I just dislike the nesting. Probably a better thing to have in this case though.